OsmAnd/src/net/osmand/plus/views/mapwidgets/configure/WidgetsSettingsHelper.java

```diff
@@ -0,0 +1,205 @@
+package net.osmand.plus.views.mapwidgets.configure;
+
+import net.osmand.plus.OsmandApplication;
+import net.osmand.plus.activities.MapActivity;
+import net.osmand.plus.settings.backend.ApplicationMode;
+import net.osmand.plus.settings.backend.OsmandSettings;
+import net.osmand.plus.settings.backend.preferences.OsmandPreference;
+import net.osmand.plus.views.mapwidgets.MapWidgetInfo;
+import net.osmand.plus.views.mapwidgets.MapWidgetRegistry;
+import net.osmand.plus.views.mapwidgets.MapWidgetsFactory;
+import net.osmand.plus.views.mapwidgets.WidgetType;
+import net.osmand.plus.views.mapwidgets.WidgetsPanel;
+import net.osmand.plus.views.mapwidgets.widgets.MapWidget;
+import net.osmand.util.Algorithms;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Set;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+
+import static net.osmand.plus.views.mapwidgets.MapWidgetRegistry.AVAILABLE_MODE;
+import static net.osmand.plus.views.mapwidgets.MapWidgetRegistry.ENABLED_MODE;
+
+public class WidgetsSettingsHelper {
+
+	private final OsmandSettings settings;
+	private final MapActivity mapActivity;
+	private final ApplicationMode appMode;
+
+	private final MapWidgetRegistry widgetRegistry;
+	private final MapWidgetsFactory widgetsFactory;
+
+	public static void resetConfigureScreenSettings(@NonNull MapActivity mapActivity,
```

Comment:
1) Why do we need this class MapWidgetsFactory is very small class and much better naming with than helper
2) why so many static methods? They all could be part of constructor and normal methods, it's much better to not wrap around static methods

Reply:
1. I don't understand question